src/ConfigEditor/Auth/auth-method/AuthMethodSettings.tsx

```diff
@@ -75,9 +77,22 @@ export const AuthMethodSettings: React.FC<Props> = ({
         return acc;
       }, {}) ?? {};
 
+    const preparedDefaultOptions = {} as Record<AuthMethod, SelectableValue<AuthMethod>>;
+    let k: keyof typeof AuthMethod;
+    for (k in defaultOptions) {
+      if (extendedDefaultOptions && extendedDefaultOptions[k]) {
+        preparedDefaultOptions[k] = {
+          ...defaultOptions[k],
+          ...extendedDefaultOptions[k],
+        }
+      } else {
+        preparedDefaultOptions[k] = defaultOptions[k];
+      }
+    }
```

Comment:
If you make `extendedDefaultOptions = extendedDefaultOptions || {}` then you can do:

``` 
preparedDefaultOptions[k] = {
          ...defaultOptions[k],
          ...extendedDefaultOptions[k],
        }```

Without needing that conditional

Reply:
Okay I'll merge it as it is and follow up later as I'm kinda worried this will throw error when I try to use it later ;)